templates/user/auth/signin_inner.tmpl

```diff
@@ -69,5 +69,22 @@
 		</div>
 	</div>
 	{{end}}
+	{{if .SAMLProviders}}
+	<div class="divider divider-text">
+		{{.locale.Tr "sign_in_or"}}
+	</div>
+	<div id="saml-login-navigator" class="gt-py-2">
+		<div class="gt-df gt-fc gt-jc">
+			<div id="saml-login-navigator-inner" class="gt-df gt-fc gt-fw gt-ac gt-gap-3">
+				{{range $provider := .SAMLProviders}}
+					<a class="{{$provider.Name}} ui button gt-df gt-ac gt-jc gt-py-3 saml-login-link" href="{{AppSubUrl}}/user/saml/{{$provider.Name}}">
```

Comment:
Maybe already start using Tailwind now that we have it?

Reply:
#29357 was just merged, i personally would not start refactorung in a feature pull.

If it would be a complete new template and somebody do create a pull against this feature branch i would merge (apply) it here.

But this file here should then be refactored as ahole and thats pull do adress a diverent issue already ;)

Reply:
You have a point, let's leave it to some later point :)